src/utils/queue.ts

```diff
@@ -0,0 +1,44 @@
+export interface QueueItem<T = any> {
+    fn: () => Promise<T>;
+    resolve: (value: T) => void;
+    reject: (reason?: any) => void;
+}
+
+let intervalId: NodeJS.Timeout | undefined = undefined;
+let firstItemInQueue: boolean = true;
+const requestQueue: QueueItem[] = [];
```

Comment:
just noting here that for our use case this is okay but generally a new queue is created in the module where you need it. if we were to use this in different providers and we supported enabling multiple providers in the same app the requests of different providers would get mixed here.